src/search/parser_tests.h

```diff
@@ -0,0 +1,210 @@
+        for (const auto& search_AST : searches) {
+            parser::Any search_any = search_AST->construct();
+            std::shared_ptr<SearchEngine> search = parser::any_cast<std::shared_ptr<SearchEngine>>(search_any);
+            search->dump(indent + "    ");
+        }
+    }
+};
+
+class IteratedSearchPlugin : public parser::Plugin<SearchEngine, IteratedSearch> {
+public:
+    IteratedSearchPlugin()
+        : parser::Plugin<SearchEngine, IteratedSearch>("iterated", "Iterated Search") {;
+        add_option<std::vector<std::shared_ptr<SearchEngine>>>(
+            "searches",
+            "use preferred operators of these evaluators", "[]", parser::Bounds::unlimited(), true);
+    }
+};
+
+class BlindHeuristicPlugin : public parser::Plugin<Evaluator, BlindHeuristic> {
+public:
+    BlindHeuristicPlugin()
+        : parser::Plugin<Evaluator, BlindHeuristic>("blind", "Blind Heuristic") {
+        document_synopsis("Returns cost of cheapest action for "
+                          "non-goal states, "
+                          "0 for goal states");
+        document_language_support("action costs", "supported");
+        document_language_support("conditional effects", "supported");
+        document_language_support("axioms", "supported");
+        document_property("admissible", "yes");
+        document_property("consistent", "yes");
+        document_property("safe", "yes");
+        document_property("preferred operators", "no");
+
+        add_option<int>("repeat", "how often to check for the goal", "2K");
+        add_option<double>("time_limit", "when to stop", "2e4");
+        add_list_option<std::vector<int>>("patterns", "...", "[[1,2,3], []]");
+
+        std::vector<std::string> lp_solvers;
+        std::vector<std::string> lp_solvers_doc;
+        lp_solvers.push_back("CLP");
+        lp_solvers_doc.push_back("default LP solver shipped with the COIN library");
+        lp_solvers.push_back("CPLEX");
+        lp_solvers_doc.push_back("commercial solver by IBM");
+        lp_solvers.push_back("GUROBI");
+        lp_solvers_doc.push_back("commercial solver");
+        lp_solvers.push_back("SOPLEX");
+        lp_solvers_doc.push_back("open source solver by ZIB");
+        add_enum_option<LPSolverType>(
+            "lpsolver",
+            lp_solvers,
+            "external solver that should be used to solve linear programs",
+            "CPLEX",
+            lp_solvers_doc);
+    }
+
+    virtual parser::Options preprocess_options(const parser::Options &options) const override {
+        options.verify_list_non_empty<std::vector<int>>("patterns");
+        parser::Options copy(options);
+        std::vector<std::vector<int>> patterns = options.get_list<std::vector<int>>("patterns");
+        for (std::vector<int> &pattern : patterns) {
+            pattern.push_back(42);
+        }
+        copy.set("patterns", patterns);
+        return copy;
+    }
+};
+
+
+static EagerSearchPlugin p1;
```

Comment:
I did the change to `static class X {} my_plugin;`. With templates we could potentially have a plugin with 5 parameters like
```C++
PLUGIN(Evaluator, BlindHeuristic, "blind", "Blind Heuristic",
        document_synopsis("...");
        document_language_support("action costs", "supported");
        // ...
        add_option<int>("repeat", "how often to check for the goal", "2K");
)
```

I'm not sure this is better, though, because it is hard to see what is going on (but maybe this is not necessary).